creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +388,73 @@ class SnapbackSM {
    * }
    */
   async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet, unhealthyReplicasSet, healthyNodes }) {
-    let currentNodeHasExistingUserState = true
+    let newReplicaNodesSet = new Set()
     let i = 0
-    let newSecondary
-    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
-      // Make sure current replica set will not be chosen as potential nodes for the new replica set
-      const healthyPotentialPeers = healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
-
+    let issueReconfig = false
+    let newPrimary, newSecondary1, newSecondary2
+    while (newReplicaNodesSet.size !== NUMBER_OF_REPLICA_SET_NODES && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
```

Comment:
also, instead of building a `newReplicaNodesSet` of size = 3 always, shouldn't we just build it to `unhealthyReplicasSet.size`? if we only need 1 new one, why select 3 new ones and never use 2 of them